lib/io/gpio.hpp

```diff
@@ -34,8 +34,10 @@ class IGpioWriter {
  */
 class IGpio {
  public:
-  virtual std::optional<std::shared_ptr<IGpioReader>> getReader(const std::uint8_t pin) = 0;
-  virtual std::optional<std::shared_ptr<IGpioWriter>> getWriter(const std::uint8_t pin) = 0;
+  virtual std::optional<std::shared_ptr<IGpioReader>> getReader(const std::uint8_t pin,
+                                                                const Edge edge) = 0;
+  virtual std::optional<std::shared_ptr<IGpioWriter>> getWriter(const std::uint8_t pin,
+                                                                const Edge edge) = 0;
```

Comment:
this is the other thing, a bunch of stuff like `Edge` are defined in `hardware_gpio.hpp`, which i don't think should be there

Reply:
Where would it be if not here though?

Reply:
yes, those should be defined here as this doesn't build